engine/graphics/src/vulkan/graphics_vulkan.cpp

```diff
@@ -2572,22 +2617,50 @@ namespace dmGraphics
         Program* program_ptr = (Program*) prog;
         ShaderModule* module = program_ptr->m_VertexModule;
 
-        if (index >= program_ptr->m_VertexModule->m_UniformCount)
+        if (index >= program_ptr->m_VertexModule->m_TotalUniformCount)
         {
             module = program_ptr->m_FragmentModule;
-            index -= program_ptr->m_VertexModule->m_UniformCount;
+            index -= program_ptr->m_VertexModule->m_TotalUniformCount;
         }
 
-        if (index >= module->m_UniformCount)
+        if (index >= module->m_TotalUniformCount)
         {
             return 0;
         }
 
-        ShaderResourceBinding* res = &module->m_Uniforms[index];
-        *type = ShaderDataTypeToGraphicsType(res->m_Type);
-        *size = res->m_ElementCount;
+        uint32_t search_index = 0;
+        for (int i = 0; i < module->m_Uniforms.Size(); ++i)
+        {
+            if (IsUniformTextureSampler(module->m_Uniforms[i]))
+            {
+                if (search_index == index)
+                {
+                    ShaderResourceBinding* res = &module->m_Uniforms[index];
+                    *type = ShaderDataTypeToGraphicsType(res->m_Type);
+                    *size = res->m_ElementCount;
+                    return (uint32_t)dmStrlCpy(buffer, res->m_Name, buffer_size);
+                }
+                search_index++;
+            }
+            else
+            {
+                for (int j = 0; j < module->m_Uniforms[i].m_BlockMembers.Size(); ++j)
+                {
+                    if (search_index == index)
+                    {
+                        UniformBlockMember& member = module->m_Uniforms[i].m_BlockMembers[j];
+                        *type = ShaderDataTypeToGraphicsType(member.m_Type);
+                        *size = member.m_ElementCount;
+                        return (uint32_t) dmStrlCpy(buffer, member.m_Name, buffer_size);
+                    }
+
+                    search_index++;
+                }
+            }
+        }
```

Comment:
The API for querying locations are based on OpenGL where all uniforms are listed as an increasing index. This might change when we have uniform buffer support for OpenGL.